Sources/StreamChat/Query/Sorting/SortValue.swift

```diff
@@ -10,49 +10,54 @@ struct SortValue<T> {
-                let lhsValue = lhs[keyPath: sort.keyPath]
-                let rhsValue = rhs[keyPath: sort.keyPath]
-
-                if sort.isAscending, evaluate(lhs: lhsValue, rhs: rhsValue, isAscending: sort.isAscending) {
-                    return true
-                } else if !sort.isAscending, evaluate(lhs: lhsValue, rhs: rhsValue, isAscending: sort.isAscending) {
-                    return true
-                } else {
-                    return false
+    /// Returns the elements of the sequence, sorted using the given sort values as the comparison between elements.
+    ///
+    /// The first `SortValue` is the primary key and subsequent `SortValue`s are for breaking the tie.
+    func sorted(using sortValues: [SortValue<Element>]) -> [Element] {
+        guard !sortValues.isEmpty else { return self }
+        return sorted { lhs, rhs in
+            for sortValue in sortValues {
+                let lhsValue = lhs[keyPath: sortValue.keyPath]
+                let rhsValue = rhs[keyPath: sortValue.keyPath]
+                let isAscending = sortValue.isAscending
+                // These are type-erased, therefore we need to manually cast them
+                if let result = nilComparison(lhs: lhsValue, rhs: rhsValue, isAscending: isAscending) {
+                    return result
+                } else if let result = areInIncreasingOrder(lhs: lhsValue, rhs: rhsValue, type: String.self, isAscending: isAscending) {
+                    return result
+                } else if let result = areInIncreasingOrder(lhs: lhsValue, rhs: rhsValue, type: Int.self, isAscending: isAscending) {
+                    return result
+                } else if let result = areInIncreasingOrder(lhs: lhsValue, rhs: rhsValue, type: Double.self, isAscending: isAscending) {
+                    return result
+                } else if let result = areInIncreasingOrder(lhs: lhsValue, rhs: rhsValue, type: Date.self, isAscending: isAscending) {
+                    return result
+                } else if let lBool = lhs as? Bool, let rBool = rhs as? Bool, lBool != rBool {
+                    // Backend considers boolean sorting in reversed order.
+                    return isAscending ? lBool && !rBool : !lBool && rBool
                 }
             }
+            return false
         }
-
-        return result
     }
-
-    private func evaluate(lhs: Any?, rhs: Any?, isAscending: Bool) -> Bool {
-        if lhs == nil, rhs != nil, !isAscending {
-            return true
-        } else if lhs != nil, rhs == nil, isAscending {
-            return true
+    
+    /// Dedicated nil handling for Any type which returns false if we do `lhs == nil`.
+    private func nilComparison(lhs: Any, rhs: Any, isAscending: Bool) -> Bool? {
+        func isAnyNil(_ value: Any) -> Bool {
+            let mirror = Mirror(reflecting: value)
+            return mirror.displayStyle == .optional && mirror.children.isEmpty
         }
```

Comment:
Is there a better way?

Reply:
can't we do something like this instead:
```
func isAnyNil(_ value: Any) -> Bool {
     return (value as Any?) == nil
}
```

Reply:
This also does not work. 

```swift
func test() {
    struct Item {
        let index: Int
        let date: Date?
    }

    let item1 = Item(index: 1, date: nil)
    let item2 = Item(index: 2, date: Date())
    let keyPath: PartialKeyPath<Item> = \.date

    let lhsValue = item1[keyPath: keyPath]
    let rhsValue = item2[keyPath: keyPath]

    print(lhsValue, rhsValue)
    print(isAnyNil(lhsValue), isAnyNil(rhsValue))
    print((lhsValue as Any?) == nil, (rhsValue as Any?) == nil)
}

func isAnyNil(_ value: Any) -> Bool {
    let mirror = Mirror(reflecting: value)
    return mirror.displayStyle == .optional && mirror.children.isEmpty
}
```
gives
```swift
nil Optional(2024-09-17 10:18:17 +0000)
true false
false false
```

Reply:
@laevandus You can check `DebugObjectViewController`, we do nil comparison there.

Reply:
I want to avoid reflection though, this will be run for every channel so it might impact performance.